### PR TITLE
Wallet - Watched Account UI Updates

### LINKED ISSUE #18105: Account switcher updates for watched account
# Feature Issue

- Add support for watched account type in the account avatar displayed in the navigation bar of the account page
- Add the watched account (with the eye icon) UI in the account switcher bottom sheet

| Account avatar in nav bar | Switcher sheet  | 
| --- | --- | 
|  |  | 

### Design

[Navigation bar](https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=1869%3A461375&mode=design&t=7owVkBgA0r53OI2V-1) 

[Switcher sheet](https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=1348%3A320994&mode=design&t=7owVkBgA0r53OI2V-1)

### User Story

> As a user, I want the watched account in the account switcher bottom sheet to be distinguished, so that I can identify and select it. 

### Acceptance Criteria

- Reuse existing subscription
- The account switcher in the navigation bar should display the account avatar with the watched account type for the watched account

### PR DESCRIPTION
**Pull request by smohamedjavid**
fixes #18105

### Summary

This PR:

- Updates the account switcher in the `page-nav` component to use `account-avatar` instead of `dropdown`
- Updates the account avatar (on the top right corner of the account screen) to display the watched account variant correctly
- Updates the accounts list in the account switcher to display the correct background for the watched account with the eye icon

 | Account Screen   | Account Switcher |
| --- | --- | 
|  |  |

### Platforms

- Android
- iOS 

#### Prerequisite: Create multiple wallet accounts (a combination of generated and watched accounts)

- Open Status
- Navigate to the Wallet tab
- Open a watched account from the accounts card slider
- Verify the account avatar on the top right corner is displayed with a background color of less opacity (watched account variant)
- Tap on the account avatar to open the account switcher
- Verify the watched account displayed in the accounts list has a different background opacity than others.

status: ready

### CODE REVIEW

File: src/quo/components/list_items/account/view.cljs [hunk 1]
```diff
@@ -11,8 +11,7 @@
     [reagent.core :as reagent]))
 
 (defn- account-view
-  [{:keys [account-props title-icon? blur? theme]
-    :or   {title-icon? false}}]
+  [{:keys [account-props title-icon blur? theme]}]
```
Comment (J-Son89) [inline src/quo/components/list_items/account/view.cljs:14]: nice! also just an observation about the removed code I thought I would share here.
In general we should have booleans as false by default. i.e they are something you have to activate not disable 👍

File: src/quo/components/list_items/account/view.cljs [hunk 2]
```diff
@@ -22,15 +21,14 @@
...
-          :color (if blur?
-                   colors/white-opa-40
-                   (colors/theme-colors colors/neutral-50 colors/neutral-40 theme))}]])]
+     (when title-icon
+       [icon/icon title-icon
+        {:size                20
+         :color               (if blur?
+                                colors/white-opa-40
+                                (colors/theme-colors colors/neutral-50 colors/neutral-40 theme))
+         :container-style     style/keycard-icon-container
```
Comment (J-Son89) [inline src/quo/components/list_items/account/view.cljs]: rename style to `style/title-icon-container` ??

File: src/status_im2/contexts/wallet/common/account_switcher/view.cljs [hunk 3]
```diff
@@ -17,8 +17,8 @@
     :or   {icon-name           :i/close
            accessibility-label :top-bar
            switcher-type       :account-options}}]
-  (let [{:keys [color emoji]} (rf/sub [:wallet/current-viewing-account])
-        networks              (rf/sub [:wallet/network-details])]
+  (let [{:keys [color emoji watch-only?]} (rf/sub [:wallet/current-viewing-account])
+        networks                          (rf/sub [:wallet/network-details])]
```
Comment (J-Son89) [inline src/status_im2/contexts/wallet/common/account_switcher/view.cljs:21]: any chance you could add the one liner for testnet in the :wallet/network-details sub here?

i.e 
_src/status_im2/subs/wallet/networks.cljs_
```
(re-frame/reg-sub
 :wallet/filtered-networks-by-mode
 :<- [:wallet/networks]
 :<- [:profile/test-networks-enabled?]
 (fn [networks test-networks-enabled?]
   (get networks (if test-networks-enabled? :test :prod))))
```
or should we do that in a separate pr?

File: src/quo/components/list_items/account/view.cljs (hunk 1 above)
Comment (smohamedjavid) [reply src/quo/components/list_items/account/view.cljs:14]: True that!

--

Adding context on why the boolean prop was replaced:

In the old code, we had `title-icon?` a boolean. If it's true, the component displays the keycard icon. 

We need the flexibility to display the icon based on the context. So, this PR replaces `title-icon?` with `title-icon` that receives the icon name and displays.

File: src/status_im2/contexts/wallet/common/account_switcher/view.cljs (hunk 3 above)
Comment (smohamedjavid) [reply src/status_im2/contexts/wallet/common/account_switcher/view.cljs:21]: I would advise creating a separate PR as this sub is not used in any screens yet and can be quickly merged.

**Review by briansztamfater — APPROVED** [review]
Looks good to me!

**Review by ulisesmac — APPROVED** [review]
Looks very good :+1:

File: src/status_im2/contexts/wallet/common/account_switcher/view.cljs (hunk 3 above)
Comment (J-Son89) [reply src/status_im2/contexts/wallet/common/account_switcher/view.cljs:21]: Sounds good! 👌

**Review by J-Son89 — APPROVED** [review]

File: src/quo/components/drawers/drawer_top/view.cljs [hunk 4]
```diff
@@ -149,22 +150,31 @@
...
+      {:size   :heading-2
+       :weight :semi-bold}
+      title]
+     (when title-icon
+       [icons/icon title-icon
+        {:container-style style/title-icon
+         :size            20
+         :color           (if blur?
+                            colors/white-opa-40
+                            (colors/theme-colors colors/neutral-50 colors/neutral-40 theme))}])]))
```
Comment (OmarBasem) [inline src/quo/components/drawers/drawer_top/view.cljs:171]: worth moving to style file?

**Review by OmarBasem — APPROVED** [review]

**Review by vkjr — APPROVED** [review]

**Comment by qoqobolo** [comment]
Thanks @smohamedjavid, looks good to me!
Please send PR to a design review if needed.

**Comment by smohamedjavid** [comment]
> Thanks @smohamedjavid, looks good to me! Please send PR to a design review if needed.

Thanks for testing the PR! @qoqobolo

The design review is not required as the individual components are reviewed. This PR updates the right UI type for the watched account which was not done initially.

The screen-level design review is yet to be done. We are planning to do one for the whole wallet before the `v1.27.0` release. 👍

**Comment by J-Son89** [comment]
@smohamedjavid - design review probably for after 1.27 because functionality first and foremost! 😃

**Comment by smohamedjavid** [comment]
> @smohamedjavid - design review probably for after 1.27 because functionality first and foremost! 😃

I stand corrected 😃
